14_ComputeReduce/main.cpp

```diff
@@ -0,0 +1,281 @@
+#include "nbl/application_templates/BasicMultiQueueApplication.hpp"
+#include "nbl/application_templates/MonoAssetManagerAndBuiltinResourceApplication.hpp"
+
+#include <chrono>
+#include <random>
+
+using namespace nbl;
+using namespace core;
+using namespace asset;
+using namespace system;
+using namespace video;
+
+class ComputeScanApp final : public application_templates::BasicMultiQueueApplication, public application_templates::MonoAssetManagerAndBuiltinResourceApplication
+{
+	using device_base_t = application_templates::BasicMultiQueueApplication;
+	using asset_base_t = application_templates::MonoAssetManagerAndBuiltinResourceApplication;
+
+public:
+	ComputeScanApp(const path& _localInputCWD, const path& _localOutputCWD, const path& _sharedInputCWD, const path& _sharedOutputCWD) :
+		system::IApplicationFramework(_localInputCWD, _localOutputCWD, _sharedInputCWD, _sharedOutputCWD) {}
+
+	bool onAppInitialized(smart_refctd_ptr<ISystem>&& system) override
+	{
+		if (!device_base_t::onAppInitialized(std::move(system)))
+			return false;
+		if (!asset_base_t::onAppInitialized(std::move(system)))
+			return false;
+
+		computeQueue = getComputeQueue();
+
+		// Create (an almost) 128MB input buffer
+		constexpr auto in_size = 128u << 10u;
+		constexpr auto in_count = in_size / sizeof(uint32_t) - 23u;
+
+		m_logger->log("Input element count: %d", ILogger::ELL_PERFORMANCE, in_count);
+
+		inputData = new uint32_t[in_count];
+		{
+			std::random_device random_device;
+			std::mt19937 generator(random_device());
+			std::uniform_int_distribution<uint32_t> distribution(0u, ~0u);
+			for (auto i = 0u; i < in_count; i++)
+				inputData[i] = distribution(generator) % 100000;
```

Comment:
you might want to make these small enough so the reduction or scan doesn't overflow (I think our algo is not necessarily resillient to that)